content/posts/xapi-activities-101/index.md

```diff
@@ -0,0 +1,262 @@
+(LRS, like [ralph](https://github.com/openfun/ralph)) and
+[Learning Record Consumers](https://github.com/adlnet/xAPI-Spec/blob/master/xAPI-About.md#def-learning-record-consumer)
+(LRC, like [warren](https://github.com/openfun/warren)) We'll try to make the concept of activity
+clearer for all parties so that anyone can make enlightened decisions about how to share or query
+activity definitions and metadata!
+
+### TL;DR
+
+The term `activity` appears in different sections of the specification and the different concepts
+introduced by the spec are somewhat intertwined with one another. To make the following explanations
+easier to grasp, we'll present them step by step according to the masterful plan below :
+
+- What is an activity according to the spec?
+- How is an activity modeled? That is, what is its structure?
+- Where can we find an activity's definition or additional metadata?
+- How do the different activity definitions source interact with each other to produce a correct
+  activity representation?
+
+## The xAPI activity ⚽
+
+Let's start with a philosophical question: what exactly is life? ... Sorry I meant, what exactly is
+an activity?
+
+According to the spec, an activity is
+[_a type of Object making up the "this" in "I did this"_](https://github.com/adlnet/xAPI-Spec/blob/master/xAPI-About.md#def-activity).
+_In the Statement "Anna tried a cake recipe 🎂", the recipe constitutes the Activity in terms of the
+xAPI Statement._
+
+From a learning record producer (LRP) point of view, activities will often solely exist in xAPI
+statements ; statements being the only way to transmit activities to a Learning Record Store (LRS).
+We'll see that from an LRS or LRC point of view, things can get more involved.
+
+## The activity recipe 🥮
+
+An activity is **always** expressed following the structure of an
+[**activity definition**](https://github.com/adlnet/xAPI-Spec/blob/master/xAPI-Data.md#activity-definition)
+
+The activity definition must not be confused with the concept of activity
+[metadata](https://github.com/adlnet/xAPI-Spec/blob/master/xAPI-Data.md#30-metadata) even though
+both terms are overlapping. The spec does not express this difference precisely but here it is in
+plain, technical english :
+
+- an activity definition describes the **structure** of an activity. The activity definition is
+  mentioned in the spec in the detail when
+  [describing the statement structure](https://github.com/adlnet/xAPI-Spec/blob/master/xAPI-Data.md#activity-definition)
+- activity [metadata](https://github.com/adlnet/xAPI-Spec/blob/master/xAPI-Data.md#30-metadata) is a
+  term that indicates that activity metadata is hosted by an external source **but must also follow
+  the activity definition structure**. In the last part we'll see that the LRS must maintain an
+  internal canonical activity definition 👾 where it MAY merge different sources of activity
+  definition (from multiple statements and/or a metadata provider)
```

Comment:
Can you provide examples (_i.e._ code snippets)?

Reply:
Adding an example here is not easy but I added 3 different statement examples throughout the post !